bootstrap/src/main/java/org/jboss/windup/bootstrap/commands/windup/ToolingModeCommand.java

```diff
@@ -0,0 +1,209 @@
+            .run(data.input, data.output, data.sourceMode, 
+            data.ignoreReport, data.ignorePatterns, data.windupHome,
+            data.source, data.target, data.customRules);
+    }
+
+    @Override
+    public CommandPhase getPhase()
+    {
+        return null;
+    }
+
+    public static boolean isToolingMode(List<String> arguments)
+    {
+        return arguments.contains(ToolingModeCommand.COMMAND_ID);
+    }
+
+    private ToolingModeData loadData()
+    {
+        ToolingModeData data = new ToolingModeData();
+        data.input = this.getInput();
+        data.output = this.getOutput();
+        data.sourceMode = this.isSourceMode();
+        data.ignoreReport = this.ignoreReport();
+        data.ignorePatterns = this.getIgnorePatterns();
+        data.windupHome = this.getWindupHome();
+        data.source = this.getSource();
+        data.target = this.getTarget();
+        data.customRules = this.getCustomRules();
+        return data;
+    }
+
+
+    private Set<String> getInput()
+    {
+        int index = arguments.indexOf(toArg("input")) + 1;
+        return Sets.toSet(this.getValues(index));
+    }
+
+    private String getOutput()
+    {
+        int outputIndex = arguments.indexOf(toArg("output")) + 1;
+        String output = arguments.get(outputIndex);
+        System.out.println(output);
+        return String.valueOf(output);
+    }
+
+    private boolean isSourceMode()
+    {
+        return this.arguments.contains(toArg(IOptionKeys.sourceModeOption));
+    }
+
+    private boolean ignoreReport()
+    {
+        return this.arguments.contains(toArg(IOptionKeys.skipReportsRenderingOption));
+    }
+
+    private List<String> getIgnorePatterns()
+    {
+        int index = arguments.indexOf(toArg("ignorePattern")) + 1;
+        return this.getValues(index);
+    }
+
+    private String getWindupHome()
+    {
+        int homeIndex = arguments.indexOf(toArg("windupHome")) + 1;
+        String home = arguments.get(homeIndex);
+        System.out.println(home);
+        return String.valueOf(home);
+    }
+
+    private List<String> getSource()
+    {
+        int index = arguments.indexOf(toArg("source")) + 1;
+        return this.getValues(index);
+    }
+
+    private List<String> getTarget()
+    {
+        int index = arguments.indexOf(toArg("target")) + 1;
+        return this.getValues(index);
+    }
+
+    private List<File> getCustomRules()
+    {
+        int index = arguments.indexOf(toArg("rules")) + 1;
+        List<File> rules = Lists.newArrayList();
+        List<String> values = this.getValues(index);
+        values.forEach(value -> rules.add(new File(value)));
+        return rules;
+    }
+
+    private String toArg(String name) {
+        return "--" + name;
+    }
+
+    private List<String> getValues(int index) {
+        List<String> values = Lists.newArrayList();
+        if (index == 0) {
+            return values;
+        }
+        while (index < arguments.size())
```

Comment:
I see that in the end index == 0 and index == 1 with arguments.size == 1 will do the same ..... is that expected ?

Reply:
hummmm, you are always skipping the 1st argument, is that expected ?

Reply:
Yes, we skip --toolingMode